web/packages/teleterm/src/ui/ConnectMyComputer/permissions.ts

```diff
@@ -0,0 +1,30 @@
+import { Cluster } from 'teleterm/services/tshd/types';
+import { ConfigService } from 'teleterm/services/config';
+import { RuntimeSettings } from 'teleterm/mainProcess/types';
+import { routing } from 'teleterm/ui/uri';
+
+/**
+ * Checks if Connect My Computer can be used for the given root cluster.
+ *
+ * The root cluster is required because `loggedInUser` and `features` are not fully defined for leaves.
+ * */
+export function isConnectMyComputerPermittedForRootCluster(
+  rootCluster: Cluster,
+  configService: ConfigService,
+  runtimeSettings: RuntimeSettings
+): boolean {
+  if (routing.ensureRootClusterUri(rootCluster.uri) !== rootCluster.uri) {
```

Comment:
What do you think about this:

```suggestion
  if (rootCluster.leaf) {
```

Reply:
Yeah, we can do this.